R/app.R

```diff
@@ -212,7 +217,10 @@ shinyAppDir_serverR <- function(appDir, options=list()) {
     if (file.exists(file.path.ci(appDir, "global.R")))
       sourceUTF8(file.path.ci(appDir, "global.R"))
   }
-  onEnd <- function() {
+  onStop <- function() {
+    onStopFunc <- getShinyOption("onStop", default = function() {})
+    if (!is(onStopFunc, "function")) warning("`onStop` must be a function")
```

Comment:
Using `is.function` would be more idiomatic. But it probably makes more sense to check the type of the object when it's registered, either in `onStop` or in `Callbacks$register`.